extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java

```diff
@@ -583,4 +604,193 @@ private void memoryPoolMetrics(MetricRegistry registry, List<String> names) {
         }
     }
 
+    private void micrometerJvmGcMetrics(MetricRegistry registry, List<String> names, ShutdownContext shutdown) {
+        if (!ImageInfo.inImageCode()) {
```

Comment:
I was just about to ask how the code knows to not expose metrics in GraalVM mode and stumbled upon this before I could :).

:+1: